packages/calling/calling/src/CallingClient/line/index.ts

```diff
@@ -0,0 +1,463 @@
+import {Mutex} from 'async-mutex';
+import {
+  ALLOWED_SERVICES,
+  HTTP_METHODS,
+  IDeviceInfo,
+  IpInfo,
+  MobiusDeviceId,
+  MobiusServers,
+  MobiusStatus,
+  RegionInfo,
+  ServiceIndicator,
+  WebexRequestPayload,
+} from '../../common/types';
+import {ICallSettingResponse, ILine, LINE_EVENTS, LineEventTypes, LineStatus} from './types';
+import {
+  LINE_FILE,
+  CALLING_USER_AGENT,
+  CISCO_DEVICE_URL,
+  DISCOVERY_URL,
+  GET_MOBIUS_SERVERS_UTIL,
+  IP_ENDPOINT,
+  KEEPALIVE_UTIL,
+  SPARK_USER_AGENT,
+  URL_ENDPOINT,
+  REGISTRATION_FILE,
+  NETWORK_FLAP_TIMEOUT,
+} from '../constants';
+import log from '../../Logger';
+import {IRegistration} from '../registration/types';
+import {createRegistration} from '../registration';
+import {ClientRegionInfo, ISDKConnector, WebexSDK} from '../../SDKConnector/types';
+import {CallingClientConfig} from '../types';
+import {Eventing} from '../../Events/impl';
+import {LineError} from '../../Errors/catalog/LineError';
+import {METRIC_EVENT, REG_ACTION, METRIC_TYPE} from '../metrics/types';
+import {LOGGER, LogContext} from '../../Logger/types';
+import {handleRegistrationErrors, filterMobiusUris, validateServiceData} from '../../common';
+import SDKConnector from '../../SDKConnector';
+import {ICallManager} from '../calling/types';
+
+export default class Line extends Eventing<LineEventTypes> implements ILine {
+  #webex: WebexSDK;
+
+  #mutex: Mutex;
+
+  #registration: IRegistration;
+
+  #sdkConfig: CallingClientConfig | undefined;
+
+  #sdkConnector: ISDKConnector;
+
+  #callManager: ICallManager;
+
+  public userId: string;
+
+  public clientDeviceUri: string;
+
+  public lineId: string;
+
+  public phoneNumber: string;
+
+  public extension: string;
+
+  public status: LineStatus;
+
+  public sipAddresses: string[];
+
+  public voicemail: string;
+
+  public lastSeen: string;
+
+  public keepaliveInterval: number;
+
+  public callKeepaliveInterval: number;
+
+  public rehomingIntervalMin: number;
+
+  public rehomingIntervalMax: number;
+
+  public voicePortalNumber: number;
+
+  public voicePortalExtension: number;
+
+  #primaryMobiusUris: string[];
+
+  #backupMobiusUris: string[];
+
+  constructor(
+    userId: string,
+    clientDeviceUri: string,
+    lineId: string,
+    phoneNumber: string,
+    extension: string,
+    status: LineStatus,
+    sipAddresses: string[],
+    voicemail: string,
+    lastSeen: string,
+    keepaliveInterval: number,
+    callKeepaliveInterval: number,
+    rehomingIntervalMin: number,
+    rehomingIntervalMax: number,
+    voicePortalNumber: number,
+    voicePortalExtension: number,
+    callManager: ICallManager,
+    mutex: Mutex,
+    config?: CallingClientConfig
+  ) {
+    super();
+    this.userId = userId;
```

Comment:
We need one more field here in the line object : MobiusDeviceId. We are going to maintain lineId and MobiusDeviceId both.

Reply:
Done